viewcontroller/viewcontroller.go

```diff
@@ -108,48 +100,61 @@ func NewRolloutViewController(rolloutId string, noColor bool) *RolloutViewContro
 	}
 }
 
-func (c *RolloutViewController) GetRollout() (rollout.DetailedRollout, error) {
-	detailedRollout := rollout.DetailedRollout{}
+func (c *RolloutViewController) GetRollout() (*rollout.DetailedRollout, error) {
+	detailedRollout := &rollout.DetailedRollout{}
+	errors := make(chan error)
+	wg := &sync.WaitGroup{}
+	wg.Add(3)
 
 	if c.previousRollout == nil {
-		strategy, err := c.rolloutRepository.GetStrategy(c.rolloutId)
-		if err != nil {
-			return detailedRollout, err
-		}
-		detailedRollout.Definition.Strategy = strategy
+		wg.Add(1)
+		go func(wg *sync.WaitGroup, errors chan error) {
+			defer wg.Done()
+			strategy, err := c.rolloutRepository.GetStrategy(c.rolloutId)
+			detailedRollout.Definition.Strategy = strategy
+			if err != nil {
+				errors <- err
+			}
+		}(wg, errors)
 	} else {
 		detailedRollout.Definition.Strategy = c.previousRollout.Definition.Strategy
 	}
 
-	fetchedRollout, err := oceancd.GetRollout(c.rolloutId)
-	if err != nil {
-		return detailedRollout, err
-	}
-
-	phases, err := oceancd.GetRolloutPhases(c.rolloutId)
-	if err != nil {
-		return detailedRollout, err
-	}
+	go func(wg *sync.WaitGroup, errors chan error) {
```

Comment:
can we extract these anonymous functions into methods with names so we can understand what they should do?

Reply:
yes

Reply:
lets do it :)